### ISSUE TITLE
SLX-OS support

### ISSUE THREAD

**Issue by wxcafe** [ytti/oxidized#1027]
So I started writing a model for SLX-OS (based on the IronWare one) and it kinda works, but doesn't go further than `show system`, then it timeouts : 

```
Sep 14 15:40:53  oxidized[1640]: Jobs 0, Want: 1
Sep 14 15:40:54  oxidized[1640]: SSH:  @ es2.par5
Sep 14 15:40:55  oxidized[1640]: SSH: terminal length 0 @ es2.par5
Sep 14 15:40:55  oxidized[1640]: SSH: show version @ es2.par5
Sep 14 15:40:56  oxidized[1640]: SSH: show chassis @ es2.par5
Sep 14 15:40:57  oxidized[1640]: SSH: show system @ es2.par5
Sep 14 15:41:17  oxidized[1640]: 10.100.1.7 raised Timeout::Error with msg "execution expired"
Sep 14 15:41:17  oxidized[1640]: es2.par5 status no_connection, retries exhausted, giving up
```

I've got no idea why though, so if you could help it would be greatly appreciated. Attached the model file I've written up so far (I figure it's not a PR, but it's better than an issue, right? :p)

[slxos.rb](https://github.com/ytti/oxidized/files/1303215/slxos.txt)

cc @afenioux

**Comment by danilopopeye** [comment]
@wxcafe could you enable the input `debug` and check where the timeout is raised. can you also provide your config?

> (I figure it's not a PR, but it's better than an issue, right?

yeah, it is 👏

**Comment by wxcafe** [comment]
What do you mean by "enable the input `debug`"? if this is about the `-d` switch to oxidized, the log I show in my previous comment is the result of having it enabled. 

I can't give you the entire config, but I can give you the part that configures the SSH daemon. I figure it's pretty normal but 🤷‍♂️ 

```
slx9850-lab# show ru | inc ssh
ssh server key rsa 2048
ssh server key ecdsa 256
ssh server key dsa
ssh server use-vrf default-vrf
ssh server use-vrf mgmt-vrf
```

The weird thing is that when I run it by hand, it seems to work fine. Yes, `show system` takes two/three seconds to answer but it shouldn't be any problem

**Comment by mikaelmantel** [comment]
Thank you for the input debug.

We think we have found the problem : 

```
Welcome to the Brocade SLX-OS Software
oxidize connected from 10.120.0.202 using ssh on es2-par5
es2-par5#
es2-par5# terminal length 0
----------^
syntax error: unknown argument.
es2-par5# show version

...

PS7 is absent
PS8 is absent
PS9 is absent
PS10 is absent
--More--
```

But we don't understand why the command doesn't work, because we can execute it without problem by hand.

**Comment by mikaelmantel** [comment]
Have you an idea ?

**Comment by danilopopeye** [comment]
@mikaelmantel I'm not a network guy just some random ruby dev 😃 

Maybe @ytti or @laf can help you debug the device.

**Comment by laf** [comment]
Yeah you can give me access to the device. I'll be coming from 46.37.189.93, feel free to send me the login details to the email listed in my github profile.

**Comment by mikaelmantel** [comment]
Ok, we have found the problem : 

On Slx-os, the role "user" can not use the "terminal length 0" command.

We have created a new read-only role with access to the “terminal” command  :
```(config)# role name userfx
(config)# rule 1 action accept operation read-write role userfx command terminal
```

After using this role on oxidized, the backup works just fine.
Thanks for your time.
The attached file "slxos.rb" is valid and works fine, we create a PR for this file or you integrate it directly ?
